src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_FIGURE_COLOR = Color.WHITE.name();
+    private static final int DEFAULT_CIRCLE_RADUIS = 10;
+    private static final int NUMBER_OF_FIGURE_TYPES = 5;
+    private static final int SIDE_SIZE_LIMIT = 101;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final int paramsRandomizer1;
+    private final int paramsRandomizer2;
+    private final String randomColor;
+
+    public FigureSupplier() {
+        paramsRandomizer1 = random.nextInt(SIDE_SIZE_LIMIT);
+        paramsRandomizer2 = random.nextInt(SIDE_SIZE_LIMIT);
+        randomColor = colorSupplier.getRandomColor();
+    }
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(NUMBER_OF_FIGURE_TYPES)) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomRectangle();
+            case 2:
+                return getRandomIscTrap();
+            case 3:
+                return getRandomSquare();
+            case 4:
+                return getRandomRightTriangle();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Circle getDefaultFigure() {
+        return new Circle(DEFAULT_FIGURE_COLOR, DEFAULT_CIRCLE_RADUIS);
+    }
+
+    public Circle getRandomCircle() {
```

Comment:
For the future, if you create some methods, that you actually use only in this class with logic, that are not related to the main getRandomFigure() method,  they should be private.
Private methods should be placed after all methods. 

In the task you need to generate figure with random properties, so each side should be random. In your case you initialise paramsRandomizer1, paramsRandomizer2 and reuse it, but this is not random parameters for each type of figures. 

If you have question about the logic of the task, please ask in chat.

Reply:
![image](https://github.com/mate-academy/jv-oop-advanced/assets/114380690/b64fbbee-f3e2-42aa-84a3-608d3d6f907f)
something like this?

Reply:
![image](https://github.com/mate-academy/jv-oop-advanced/assets/114380690/adc6b8b4-ff06-4449-a438-4b7a69665a97)
something like this?

Reply:
You shouldn't create FigureSupplier figureSupplier object. If you create randomSize() method in FigureSupplier class you have an access for it without creating object.
new IsoscelesTrapezoid(randomColor, randomSize(), randomSize(), randomSize());
new Square(randomColor, randomSize());